superset/dashboards/filters.py

```diff
@@ -107,8 +121,9 @@ def apply(self, query: Query, value: Any) -> Query:
         query = query.filter(
             or_(
                 Dashboard.id.in_(owner_ids_query),
-                Dashboard.id.in_(published_dash_query),
+                Dashboard.id.in_(datesource_perm_query),
                 Dashboard.id.in_(users_favorite_dash_query),
+                Dashboard.id.in_(roles_based_query),
```

Comment:
While this logic is fairly straight forward, due to this being very critical security logic, I'd prefer to place these changes behind a FF until it has been established to be fully safe.